### PR TITLE
Basics of address question

### LINKED ISSUE #6917: [North Star Applicant UI] Implement Address Question type
#### Describe the solution presented in this issue.
Render address questions and allow users to submit, rendering any errors present. This does not cover the address correction flow

#### Done when:
What criteria need to be met for the issue to be finished? 
- [ ] Can render address questions
- [ ] User can respond to address questions
- [ ] Shows errors at question or individual field level

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Renders address question in thymeleaf. This does not change the address correction flow

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #6917

### CODE REVIEW

**Comment by greg-hyde** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/6770?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#6770** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6770?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6740** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6740?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6734** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6734?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6715** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6715?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6710** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6703** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6703?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6702** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6702?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6701** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6701?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6700** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6700?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6699** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6699?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6513** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6513?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3 other dependent PRs ([#6514](https://github.com/civiform/civiform/pull/6514) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6514?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6531](https://github.com/civiform/civiform/pull/6531) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6531?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6581](https://github.com/civiform/civiform/pull/6581) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6581?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/6770?utm_source=stack-comment).

File: server/app/views/questiontypes/AddressQuestionFragment.html [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        th:name="${addressQuestion.getStatePath()}"
+        th:aria-invalid="${!addressQuestion.getValidationErrors().isEmpty()}"
+        th:aria-required="${!question.isOptional()}"
+      >
+        <option
+          hidden
+          th:selected="${addressQuestion.getStateValue().isEmpty()}"
+          th:label="#{label.selectState}"
+        ></option>
+        <optgroup>
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/AddressQuestionFragment.html:102]: Can this `optgroup` also go away?

File: server/app/views/questiontypes/AddressQuestionFragment.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        th:text="#{label.street}"
+      ></label>
+      <input
+        class="usa-input"
+        th:id="${inputId}"
+        th:name="${addressQuestion.getStreetPath()}"
+        th:value="${addressQuestion.getStreetValue().orElse('')}"
+        th:aria-invalid="${!addressQuestion.getValidationErrors().isEmpty()}"
+        th:aria-required="${!question.isOptional()}"
+        autocomplete="address-line1"
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/AddressQuestionFragment.html:35]: TIL about autocomplete. Thanks you! :smile:

File: server/app/views/questiontypes/AddressQuestionFragment.html [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        class="usa-label"
+        th:for="${inputId}"
+        th:text="#{label.state}"
+      ></label>
+      <select
+        class="usa-select"
+        th:id="${inputId}"
+        th:name="${addressQuestion.getStatePath()}"
+        th:aria-invalid="${!addressQuestion.getValidationErrors().isEmpty()}"
+        th:aria-required="${!question.isOptional()}"
```
Comment (gwendolyngoetz) [inline server/app/views/questiontypes/AddressQuestionFragment.html:95]:
```suggestion
        th:aria-required="${!question.isOptional()}"
        autocomplete="address-level1"
```

Looks like `autocomplete` is a valid attribute for select elements. Can you include `address-level1`? If you purposefully left it out maybe leave a comment on why.

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

File: server/app/views/questiontypes/AddressQuestionFragment.html [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        autocomplete="address-line1"
+      />
+      <div
+        th:replace="~{questiontypes/QuestionBaseFragment :: validationErrors(${addressQuestion.getValidationErrors().get(addressQuestion.getStreetPath())})}"
+      ></div>
+    </th:block>
+
+    <!-- Street address line 2 -->
+    <th:block th:with="inputId=${#strings.randomAlphanumeric(8)}">
```
Comment (dkatzz) [inline server/app/views/questiontypes/AddressQuestionFragment.html:43]: **suggestion (non-blocking):** These all seem to have a similar structure - you could consider creating a AddressTextInputFragment or something that takes the labelText, inputName, inputValue and autocompleteType. If you think this ends up being clearer, that's fine though.

Comment (greg-hyde) [reply server/app/views/questiontypes/AddressQuestionFragment.html:43]: These will get changed some with the addition of more CSS classes, but It's something to consider (probably more broadly than just address) to have the equivalent fragment of FieldWithLabel. I'll just have to think about how to handle optional parameters. I'll look into it as a follow up refactor

File: server/app/views/questiontypes/AddressQuestionFragment.html (hunk 1 above)
Comment (greg-hyde) [reply server/app/views/questiontypes/AddressQuestionFragment.html:103]: done

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**